### PR TITLE
PUB-2249 - Refactor MI Endpoints to use Models

### PR DESCRIPTION
**Pull request by NatashaAlker**
JIRA link (if applicable)

https://tools.hmcts.net/jira/browse/PUB-2249

Change description

Update MI endpoints to use data models rather than strings

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/pip/account/management/controllers/CustomAccountRetrievalTest.java [hunk 1]
```diff
@@ -106,22 +108,24 @@ void testMiAccountDataRequestSuccess() throws Exception {
...
-            responseMiData.contains(createdUserId),
-            "Should successfully retrieve MI data"
+        List<AccountMiData> miData = Arrays.asList(
+            OBJECT_MAPPER.readValue(
+                responseMiData.getResponse().getContentAsString(),
+                AccountMiData[].class
+            )
         );
+
+        assertEquals(6, miData.size(), VALIDATION_MI_REPORT);
```
Comment (KianKwa) [inline src/integrationTest/java/uk/gov/hmcts/reform/pip/account/management/controllers/CustomAccountRetrievalTest.java:127]: Don't understand why is this has a size 6. Does it reply on the data from other tests in this class

File: src/main/java/uk/gov/hmcts/reform/pip/account/management/database/UserRepository.java [hunk 2]
```diff
@@ -24,10 +25,10 @@ List<PiUser> findExistingByProvenanceId(@Param("provUserId") String provenanceUs
...
     Optional<PiUser> findByUserId(UUID userId);
 
-    @Query(value = "SELECT cast(user_id as text), provenance_user_id, user_provenance, roles, created_date, "
-        + "last_signed_in_date FROM pi_user",
-        nativeQuery = true)
-    List<String> getAccManDataForMI();
+    @Query("SELECT new uk.gov.hmcts.reform.pip.model.report.AccountMiData("
+        + "u.userId, u.provenanceUserId, u.userProvenance, u.roles, u.createdDate, u.lastSignedInDate) "
+        + "FROM PiUser u")
+    List<AccountMiData> getAccManDataForMI();
```
Comment (KianKwa) [inline src/main/java/uk/gov/hmcts/reform/pip/account/management/database/UserRepository.java:31]: This is not your change but while you are there are you OK to rename the method name to something like getAccountDataForMi() or getMiData() or getAccountMiData(), etc.? 'AccMan' does not sound good :)

File: src/integrationTest/java/uk/gov/hmcts/reform/pip/account/management/controllers/CustomAccountRetrievalTest.java (hunk 1 above)
Comment (NatashaAlker) [reply src/integrationTest/java/uk/gov/hmcts/reform/pip/account/management/controllers/CustomAccountRetrievalTest.java:127]: This is an error on my part - I added it as part of debugging for the test below and forgot to remove it. Have removed now :)

File: src/main/java/uk/gov/hmcts/reform/pip/account/management/database/UserRepository.java (hunk 2 above)
Comment (NatashaAlker) [reply src/main/java/uk/gov/hmcts/reform/pip/account/management/database/UserRepository.java:31]: done :)

**Review by KianKwa — APPROVED** [review]

**Pull request merged**